src/rgw/rgw_file.cc

```diff
@@ -1355,6 +1357,250 @@ namespace rgw {
     return op_ret;
   } /* exec_continue */
 
+  int RGWFileHandle::insert_flock(flock &target, uint32_t flags)
+  {
+    list<multimap<uint64_t, flock>::iterator> overlaps, neighbors;
```

Comment:
we'd like to support efficient selection of client(s) preferred or selected to get a lock when blocking locks are released;  freebsd has a dependency graph for this, see https://github.com/freebsd/freebsd/blob/master/sys/kern/kern_lockf.c